app/models/graders/junit_grader.rb

```diff
@@ -236,4 +279,103 @@ def proper_configuration
       add_error("Could not read upload: #{e_msg}")
     end
   end
+
+  # Orca grading job methods.
+
+  # Generates a secret to be paired with an Orca grading job
+  # and compared upon response. Returns the secret and the
+  # file_path to which it was saved.
+  def generate_orca_secret!(sub)
+    grader_dir = sub.upload.grader_path(self)
+    secret = SecureRandom.hex(32) 
+    file_path = grader_dir.join("orca.secret")
+    File.open(file_path, "w") do |secret_file|
+      secret_file.write(secret)
+    end
+    return secret, file_path
+  end
+
+  def post_job_json_with_retry(uri, job_json)
```

Comment:
This logic needs to run on another thread or process, or else it will block the UI thread.